src/recipe/recipe.controller.ts

```diff
@@ -71,9 +75,9 @@ export class RecipeController {
     description: 'User is not an admin',
   })
   async getAllRecipes(
-    @Query('page') page: number,
-    @Query('limit') limit: number,
-  ) {
+    @Query('page') page: string,
+    @Query('limit') limit: string,
```

Comment:
for now, we can leave it as is (query params: string).
In general, Nest can parse payload to proper types out of the box. It's a bit weird in some cases but in most works just fine. 
You can read more about it here:
https://github.com/nestjs/nest/issues/4081#issuecomment-589640963
https://docs.nestjs.com/techniques/validation#transform-payload-objects